RELEASE/scripts/autoscend/auto_quest_level_12.ash

```diff
@@ -1290,6 +1299,223 @@ boolean L12_themtharHills()
+	adv_needed -= $location[McMillicancuddy's Other Back 40].turns_spent;
+	
+	//calculated sidequests
+	if(get_property("sidequestArenaCompleted") == faction)
+	{
+		sidequests_done++;
+	}
+	if(get_property("sidequestJunkyardCompleted") == faction)
+	{
+		sidequests_done++;
+	}
+	if(get_property("sidequestLighthouseCompleted") == faction)
+	{
+		sidequests_done++;
+	}
+	if(get_property("sidequestOrchardCompleted") == faction)
+	{
+		sidequests_done++;
+	}
+	if(get_property("sidequestNunsCompleted") == faction)
+	{
+		sidequests_done++;
+	}
+	
+	int base_kills = 2^sidequests_done;
+	int bonus_kills = 0;
+		
+	//Avatar of Sneaky Pete specific check
+	//TODO find correct spelling of mafia property for Rocket Launcher, I am currently guessing its spelling
+	if(get_property("peteMotorbikeCowling") == "Rocket Launcher")
+	{
+		bonus_kills += 3;
+	}
+	
+	//License to Adventure Path specific check
+	//TODO add it. it deals +3 kills per battle
+	
+	//adventures saved calculation
+	boolean path_need_special_handling = false;
+	if(auto_my_path() == "Pocket Familiars")
+	{
+		path_need_special_handling = true;
+	}
+	
+	if(!path_need_special_handling)
+	{
+		//hippies save 24 adv on a perfectly implemented flyering. currently they don't flyer at all. Thus it currently saves 32 adventures.
+		//fixing hippy flyering requires doing the war much sooner than it is currently done.
+		//if fixed you will lose 1 adv on doing the farm but gain early access to potentially useful food via the farm. So maybe do it anyways?
+		//also need to account for paths in which nuns or orchard are unavailable. in those paths the savings for hippies are higher.
+		if(faction == "hippy")
+		{
+			//enemiesRemaining = 1000-get_property("fratboysDefeated").to_int();
+			//enemies remaining currently does not matter for hippies
+			adv_saved = 32;
+		}
+		if(faction == "fratboy")
+		{
+			enemiesRemaining = 1000-get_property("hippiesDefeated").to_int();
+			adv_saved = ceil(enemiesRemaining.to_float() / (base_kills + bonus_kills)) - ceil(enemiesRemaining.to_float() / (2*base_kills + bonus_kills));
+		}
+	}
+	if(auto_my_path() == "Pocket Familiars")
+	{
+		//Pokefam has 500 enemies to start with with all 6 sidequests accessible. But arena and junkyard are impossible to complete
+		//TODO: find out if pokefam starts with 500 enemies defeated out of 1000 total. or 0 defeated out of 500 total
+		//current code assumes it is 0 defeated out of 500 total
+		if(faction == "hippy")
+		{
+			enemiesRemaining = 500-get_property("fratboysDefeated").to_int();
+		}
+		if(faction == "fratboy")
+		{
+			enemiesRemaining = 500-get_property("hippiesDefeated").to_int();
+		}
+		adv_saved = ceil(enemiesRemaining.to_float() / (base_kills + bonus_kills)) - ceil(enemiesRemaining.to_float() / (2*base_kills + bonus_kills));
+	}
+	
+	//is it worth it to farm a chaos butterfly?
+	if(!get_property("chaosButterflyThrown").to_boolean() && item_amount($item[chaos butterfly]) == 0)
```

Comment:
Intentionally trying to farm a chaos butterfly is very fancy. I suggest you move it into its own function though.

Reply:
Currently problematic.
almost all of the code here is just figuring out if we even want to farm it, and in order to actually move it to its own function I would need to create 2 spinoff functions farm_adv_saved() and farm_adv_to_finish() which are overly specific to farm to spin off.
when I figure out how to handle those two i will come back to this one.

Reply:
commit ef25a09